scheduler/scheduler.go

```diff
@@ -575,36 +585,39 @@ func (s *scheduler) stopTask(id, source string) []serror.SnapError {
 	default:
 		// Group depndencies by the host they live on and
 		// unsubscirbe them since task is stopping.
-		depGroupMap := s.gatherMetricsAndPlugins(t.workflow)
-
+		depGroupMap := make(map[string]depGroup)
+		depGroupMap[""] = depGroup{Metrics: []core.Metric{}}
+		getWorkflowPlugins(t.workflow.processNodes, t.workflow.publishNodes, depGroupMap)
 		var errs []serror.SnapError
 		for k := range depGroupMap {
 			mgr, err := t.RemoteManagers.Get(k)
 			if err != nil {
 				errs = append(errs, serror.New(err))
 			} else {
-				uerrs := mgr.UnsubscribeDeps(t.ID(), depGroupMap[k].Metrics, returnCorePlugin(depGroupMap[k].Plugins))
+				uerrs := mgr.UnsubscribeDeps(t.ID(), []core.Metric{}, returnCorePlugin(depGroupMap[k].Plugins))
```

Comment:
Would You mind adding some comment why `[]core.Metric{}` is used instead `depGroupMap[k].Metrics`? Thanks